internal/services/network/custom_ip_prefix_resource.go

```diff
@@ -0,0 +1,550 @@
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"commissioning_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"internet_advertising_disabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+
+		"tags": commonschema.Tags(),
+
+		"zones": commonschema.ZonesMultipleOptionalForceNew(),
+	}
+}
+
+func (r CustomIpPrefixResource) Attributes() map[string]*pluginsdk.Schema {
+	return nil
+}
+
+func (r CustomIpPrefixResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 6 * time.Hour,
+
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			r.client = metadata.Client.Network.CustomIPPrefixesClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			timeout, _ := ctx.Deadline()
+
+			var model CustomIpPrefixModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			id := parse.NewCustomIpPrefixID(subscriptionId, model.ResourceGroupName, model.Name)
+
+			existing, err := r.client.Get(ctx, id.ResourceGroup, id.CustomIpPrefixeName, "")
+			if err != nil {
+				if !utils.ResponseWasNotFound(existing.Response) {
+					return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+				}
+			}
+
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			roaValidityEndDate, err := time.Parse("2006-01-02", model.ROAValidityEndDate)
+			if err != nil {
+				return err
+			}
+			authorizationMessage := fmt.Sprintf("%s|%s|%s", subscriptionId, model.CIDR, roaValidityEndDate.Format("20060102"))
+
+			properties := network.CustomIPPrefix{
+				Name:             &model.Name,
+				Location:         pointer.To(location.Normalize(model.Location)),
+				Tags:             tags.FromTypedObject(model.Tags),
+				ExtendedLocation: nil,
+				CustomIPPrefixPropertiesFormat: &network.CustomIPPrefixPropertiesFormat{
+					Cidr:                 &model.CIDR,
+					SignedMessage:        &model.WANValidationSignedMessage,
+					AuthorizationMessage: &authorizationMessage,
+					CommissionedState:    network.CommissionedStateProvisioning,
+					NoInternetAdvertise:  &model.InternetAdvertisingDisabled,
+				},
+			}
+
+			if len(model.Zones) > 0 {
+				properties.Zones = &model.Zones
+			}
+
+			future, err := r.client.CreateOrUpdate(ctx, id.ResourceGroup, id.CustomIpPrefixeName, properties)
+			if err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			if err = future.WaitForCompletionRef(ctx, r.client.Client); err != nil {
+				return fmt.Errorf("waiting for creation of %s: %+v", id, err)
+			}
+
+			stateConf := &pluginsdk.StateChangeConf{
+				Pending:    []string{string(network.ProvisioningStateUpdating)},
+				Target:     []string{string(network.ProvisioningStateSucceeded)},
+				Refresh:    r.provisioningStateRefreshFunc(ctx, id),
+				MinTimeout: 1 * time.Minute,
```

Comment:
Given the expected (and unavoidable) long running time for this, should we use `PollInterval` here and reduce the noise?

Reply:
For this check, moving to Provisioned (the first stage after creating) is actually fairly quick and happens on the order of minutes.